cdap-app-fabric/src/main/java/io/cdap/cdap/internal/namespace/credential/handler/GcpWorkloadIdentityHttpHandler.java

```diff
@@ -0,0 +1,267 @@
+   * @throws IOException         If transport errors occur.
+   * @throws NotFoundException   If the namespace or identity are not found.
+   */
+  @GET
+  @Path("/namespaces/{namespace-id}/credentials/workloadIdentity")
+  public void getIdentity(HttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespace) throws Exception {
+    NamespaceMeta namespaceMeta = getNamespaceMeta(namespace);
+    CredentialIdentityId credentialIdentityId = createIdentityIdOrPropagate(namespace,
+        GcpWorkloadIdentityUtil.getWorkloadIdentityName(namespaceMeta.getIdentity()));
+    switchToInternalUser();
+    Optional<CredentialIdentity> identity = credentialIdentityManager.get(credentialIdentityId);
+    if (!identity.isPresent()) {
+      throw new NotFoundException("Namespace identity not found.");
+    } else {
+      NamespaceWorkloadIdentity workloadIdentity = new NamespaceWorkloadIdentity(
+          identity.get().getIdentity(), identity.get().getSecureValue());
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(workloadIdentity));
+    }
+  }
+
+  /**
+   * Creates a new identity.
+   *
+   * @param request   The HTTP request.
+   * @param responder The HTTP responder.
+   * @param namespace The identity namespace.
+   * @throws AlreadyExistsException If the identity exists.
+   * @throws BadRequestException    If the identity name or identity are invalid.
+   * @throws IOException            If transport errors occur.
+   * @throws NotFoundException      If the namespace is not found.
+   */
+  @PUT
+  @Path("/namespaces/{namespace-id}/credentials/workloadIdentity")
+  public void createIdentity(FullHttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespace) throws Exception {
+    accessEnforcer.enforce(new NamespaceId(namespace), NamespacePermission.UPDATE_SERVICE_ACCOUNT);
+    NamespaceWorkloadIdentity namespaceWorkloadIdentity =
+        deserializeRequestContent(request, NamespaceWorkloadIdentity.class);
+    if (Strings.isNullOrEmpty(namespaceWorkloadIdentity.getIdentity())) {
+      throw new BadRequestException("Identity cannot be null or empty.");
+    }
+    NamespaceMeta namespaceMeta = getNamespaceMeta(namespace);
+    validateNamespaceIdentity(namespaceMeta, namespaceWorkloadIdentity);
+    CredentialIdentityId credentialIdentityId = createIdentityIdOrPropagate(namespace,
+        GcpWorkloadIdentityUtil.getWorkloadIdentityName(namespaceMeta.getIdentity()));
+    switchToInternalUser();
+    Optional<CredentialIdentity> identity = credentialIdentityManager.get(credentialIdentityId);
+    CredentialIdentity credentialIdentity = new CredentialIdentity(
+        NamespaceId.SYSTEM.getNamespace(), GcpWorkloadIdentityUtil.SYSTEM_PROFILE_NAME,
+        namespaceMeta.getIdentity(), namespaceWorkloadIdentity.getServiceAccount());
+    if (identity.isPresent()) {
+      credentialIdentityManager.update(credentialIdentityId, credentialIdentity);
+    } else {
```

Comment:
nit: We can omit the else for readability.

Reply:
Removing else here can lead to code duplication in this case, something like:
```
if (identity.isPresent()) {
      credentialIdentityManager.update(credentialIdentityId, credentialIdentity);
 responder.sendStatus(HttpResponseStatus.OK);
 return;
} 
credentialIdentityManager.create(credentialIdentityId, credentialIdentity);
responder.sendStatus(HttpResponseStatus.OK);
```

Reply:
Oh I misread this. Please disregard, thanks!